app/knowledge_panels.py

```diff
@@ -0,0 +1,33 @@
+from typing import Union
+
+from urllib.parse import urlencode
+
+from .models import HungerGameFilter
+
+
+def hunger_game_kp(
+    hunger_game_filter: HungerGameFilter,
+    value: Union[str, None] = None,
+    country: Union[str, None] = None,
+):
+    baseUrl = "https://hunger.openfoodfacts.org/?type="
+    valueQuery = urlencode({"value_tag": value})
+    countryQuery = urlencode({"country": country})
+    if country == None and value != None:
+        html = f"<p><a href='{baseUrl}{hunger_game_filter}&{valueQuery}'></a></p>\n"
+    elif value == None and country != None:
+        html = f"<p><a href='{baseUrl}{hunger_game_filter}&{countryQuery}'></a></p>\n"
+    elif value == None and country == None:
+        html = f"<p><a href='{baseUrl}{hunger_game_filter}'></a></p>\n"
+    else:
+        html = f"<p><a href='{baseUrl}{hunger_game_filter}&{valueQuery}&{countryQuery}'></a></p>\n"
```

Comment:
You didn't get it about urlencode ;-) the idea is to de-correlate both argument handling, by treating them separately.
Also your link does not have a text so would be invisible to users.
Also note that you should alway use `== None` but instead use `is None`. (same for `!= None` ==> `is not None`.

Here is how you can separate problems:

```suggestion
    query = {}
    description = ""
    if hunger_game_filter == "country":
        # superseed country parameter
        country = value
        # and do not use facet
        hunger_game_filter = value = None
    if country is not None:
        query["country"] = country
        description = country
    if value_tag is not None:
        query["type"] = hunger_game_filter
        query["value_tag"] = value
        description = f"{value} {hunger_game_filter}"
    questions_url = "https://hunger.openfoodfacts.org/"
    if query:
        questions_url += f"?{urlencode(query)}"
    description = f"Answer robotoff questions about {description}"
    html = f"<p><a href='{questions_url}'>{description}</a></p>\n"
```

Reply:
I forgot to add that you didn't handle using country facet… the above code does this.

Reply:
As I see that in your test you even test hunger_game_filter alone, I have just updated code suggestion to enable that.